crates/nu-utils/src/filesystem.rs

```diff
@@ -34,74 +31,14 @@ pub fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
 
 #[cfg(unix)]
 pub fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
-    match dir.as_ref().metadata() {
-        Ok(metadata) => {
-            let mode = Mode::from_bits_truncate(metadata.mode() as mode_t);
-            let current_user_uid = users::get_current_uid();
-            if current_user_uid.is_root() {
-                return PermissionResult::PermissionOk;
-            }
-            let current_user_gid = users::get_current_gid();
-            let owner_user = Uid::from_raw(metadata.uid());
-            let owner_group = Gid::from_raw(metadata.gid());
-            match (
-                current_user_uid == owner_user,
-                current_user_gid == owner_group,
-            ) {
-                (true, _) => {
-                    if mode.contains(Mode::S_IXUSR) {
-                        PermissionResult::PermissionOk
-                    } else {
-                        PermissionResult::PermissionDenied(
-                            "You are the owner but do not have execute permission",
-                        )
-                    }
-                }
-                (false, true) => {
-                    if mode.contains(Mode::S_IXGRP) {
-                        PermissionResult::PermissionOk
-                    } else {
-                        PermissionResult::PermissionDenied(
-                            "You are in the group but do not have execute permission",
-                        )
-                    }
-                }
-                (false, false) => {
-                    if mode.contains(Mode::S_IXOTH)
-                        || (mode.contains(Mode::S_IXGRP)
-                            && any_group(current_user_gid, owner_group))
-                    {
-                        PermissionResult::PermissionOk
-                    } else {
-                        PermissionResult::PermissionDenied(
-                            "You are neither the owner, in the group, nor the super user and do not have permission",
-                        )
-                    }
-                }
-            }
-        }
-        Err(_) => PermissionResult::PermissionDenied("Could not retrieve file metadata"),
+    match access(dir.as_ref(), AccessFlags::X_OK) {
+        Ok(_) => PermissionResult::PermissionOk,
+        Err(Errno::EPERM) => PermissionResult::PermissionDenied("Permission denied"),
+        Err(Errno::ENOENT) => PermissionResult::PermissionDenied("Path not exists?"),
+        Err(_) => PermissionResult::PermissionDenied("Unknown error"),
```

Comment:
could you actually remove the `&str` member of `PermissionResult::PermissionDenied` while you're changing this? We never actually access it, there's no reason to even add an explanation here. after that, this can safely be reduced to 

```suggestion
    match access(dir.as_ref(), AccessFlags::X_OK) {
        Ok(_) => PermissionResult::PermissionOk,
        Err(_) => PermissionResult::PermissionDenied,
```

Reply:
> could you actually remove the `&str` member of `PermissionResult::PermissionDenied` while you're changing this? We never actually access it, there's no reason to even add an explanation here. after that, this can safely be reduced to

Indeed, I will take this after have my dinner